### PR TITLE
fix: workflow

### PR DESCRIPTION
**Pull request by septs**

### CODE REVIEW

**Review by Jack-Works — APPROVED** [review]

File: packages/cli/build/dev.ts [hunk 1]
```diff
@@ -1,29 +1,28 @@
...
 }
 
-main().then((p) => {
-    p.addListener('exit', (code) => {
-        process.exit(code)
-    })
-})
+main()
+    .then(awaitChildProcess)
+    .catch((code: number) => process.exit(code))
```
Comment (Jack-Works) [inline packages/cli/build/dev.ts:28]: Should quit if resolved normally.

Comment (septs) [reply packages/cli/build/dev.ts:28]: exit code is zero = ok

File: packages/cli/build/main.ts [hunk 2]
```diff
@@ -33,17 +34,26 @@ export default async function main(mode: 'dev' | 'build') {
...
-    if (mode === 'dev') command.unshift('serve')
-    args.filter((x) => !x.startsWith('-')).forEach((target) => {
-        command.push('--env', target)
-        if (!knownTargets.includes(target)) {
+    // prettier-ignore
+    const command = [
+        'webpack',
+        '--mode',
+        mode === 'dev' ? 'development' : 'production',
+        mode === 'dev' ? 'serve' : undefined,
```
Comment (Jack-Works) [inline packages/cli/build/main.ts:42]: Is `webpack --mode development serve` the same effect as `webpack serve --mode development`?

File: packages/cli/build/main.ts [hunk 3]
```diff
@@ -33,17 +34,26 @@ export default async function main(mode: 'dev' | 'build') {
...
 
-    const command = ['--mode', mode === 'dev' ? 'development' : 'production']
-    if (mode === 'dev') command.unshift('serve')
-    args.filter((x) => !x.startsWith('-')).forEach((target) => {
-        command.push('--env', target)
-        if (!knownTargets.includes(target)) {
+    // prettier-ignore
+    const command = [
+        'webpack',
+        mode === 'dev' ? 'serve' : undefined,
```
Comment (Jack-Works) [inline packages/cli/build/main.ts:40]: is it ok to have undefined in the argument list?

File: packages/cli/build/main.ts [hunk 4]
```diff
@@ -33,17 +34,26 @@ export default async function main(mode: 'dev' | 'build') {
...
+        if (target.startsWith('-')) {
+            continue
+        } else if (!knownTargets.includes(target)) {
             throw new TypeError(`Unknown target ${target}. Known targets: ${knownTargets}`)
         }
-    })
-    return spawn('npx', ['webpack', ...command], {
+        command.push('--env', target)
+    }
+    return spawn('npx', compact(command), {
```
Comment (septs) [inline packages/cli/build/main.ts:52]: @Jack-Works https://lodash.com/docs/#compact

**Review by Jack-Works — APPROVED** [review]

**Pull request merged**